### PR TITLE
Issue #580 : Make a method on Providers for resources.

### PR DESCRIPTION
**Pull request by tiaped**

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/Providers.java [hunk 1]
```diff
@@ -57,4 +57,27 @@ public static Provider provider(String source) {
...
+     * @param pathToResource path to resource in provided class loader. As working relatively to a class loader, no "/"
+     * at the beginning of the path
+     * @param encoding encoding
+     * @return the provider associated to pathToResource
+     */
+    public static Provider resourceProvider(ClassLoader classLoader, String pathToResource, Charset encoding) {
+        return provider(classLoader.getResourceAsStream(pathToResource), encoding);
+    }
+
+    public static Provider resourceProvider(String pathToResource, Charset encoding) {
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/Providers.java:74]: Does this and the following one need a "/"? I'm alway confused...

**Comment by matozoid** [comment]
This is a fine PR :) 

Implements #580

File: javaparser-core/src/main/java/com/github/javaparser/Providers.java (hunk 1 above)
Comment (tiaped) [reply javaparser-core/src/main/java/com/github/javaparser/Providers.java:74]: I'm far to be a master of clarity :) 
I will add javadoc to each of the methods. In order to add the precision for each one of them.
Is the first javadoc OK for you?

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/Providers.java:74]: Oh, I'm all for documenting only things that are not obvious, so you can leave out "@param encoding encoding @return the provider associated to pathToResource"

(Just assume that people only look at the Javadoc if they are confused, so it should contain information on things that may be confusing :) )

**Comment by matozoid** [comment]
Thanks @tiaped - you're welcome to pick up more issues!

**Pull request merged**